pkg/cmd/spks.go

```diff
@@ -55,142 +54,145 @@ func SpksCMD() *cli.Command {
+	billingRecords := []odoo.OdooMeteredBillingRecord{
+		{
+			ProductID:     "appcat-spks-mariadb-standard",
+			InstanceID:    "mariadb-standard",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(mariadbStandard),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
+		{
+			ProductID:     "appcat-spks-mariadb-premium",
+			InstanceID:    "mariadb-premium",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(mariadbPremium),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
+		{
+			ProductID:     "appcat-spks-redis-standard",
+			InstanceID:    "redis-standard",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(redisStandard),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
+		{
+			ProductID:     "appcat-spks-redis-premium",
+			InstanceID:    "redis-premium",
+			SalesOrder:    salesOrder,
+			UnitID:        UnitID,
+			ConsumedUnits: float64(redisPremium),
+			TimeRange: odoo.TimeRange{
+				From: startYesterdayAbsolute,
+				To:   endYesterdayAbsolute,
+			},
+		},
+	}
+	return billingRecords
+}
+
+func getDatabasesCounts(prometheusURL string, prometheusQueryArr [4]string, logger logr.Logger, startOfToday time.Time, allMetrics map[string]map[string]prometheus.Counter) (int, int, int, int) {
+
+	client, err := api.NewClient(api.Config{
+		Address: prometheusURL,
+	})
+	if err != nil {
+		logger.Error(err, "Error creating Prometheus client")
+	}
+
+	v1api := v1.NewAPI(client)
+	ctxx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	mariadbStandard := QueryPrometheus(ctxx, v1api, prometheusQueryArr[0], logger, startOfToday, allMetrics["providerMetrics"])
+
+	mariadbPremium := QueryPrometheus(ctxx, v1api, prometheusQueryArr[1], logger, startOfToday, allMetrics["providerMetrics"])
+
+	redisStandard := QueryPrometheus(ctxx, v1api, prometheusQueryArr[2], logger, startOfToday, allMetrics["providerMetrics"])
+
+	redisPremium := QueryPrometheus(ctxx, v1api, prometheusQueryArr[3], logger, startOfToday, allMetrics["providerMetrics"])
+
+	return mariadbStandard, mariadbPremium, redisStandard, redisPremium
+}
+
+func QueryPrometheus(ctx context.Context, v1api v1.API, query string, logger logr.Logger, absoluteBeginningTime time.Time, providerMetrics map[string]prometheus.Counter) int {
 	result, warnings, err := v1api.Query(ctx, query, absoluteBeginningTime, v1.WithTimeout(5*time.Second))
 	if err != nil {
+		providerMetrics["providerFailed"].Inc()
 		logger.Error(err, "Error querying Prometheus")
-		return 0, err
+		return -1
```

Comment:
Why aren't you returning `(int, error)` for this function? That would make it much clearer.

Also, instead of increasing the metrics here, you could do it in `runSPKSBilling()`. It would simplify the logic, as you'd only have to catch one error case and not all three as in this function.